gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -276,8 +277,12 @@ protected void submitFlowToDagManagerHelper(DagActionStore.DagAction dagAction,
     try {
       URI flowUri = FlowSpec.Utils.createFlowSpecUri(flowId);
       spec = (FlowSpec) flowCatalog.getSpecs(flowUri);
```

Comment:
should `FlowCatalog.getSpecs` override w/ a covariant return type?

i.e. the `FlowSpec` in:
```
  @Override
  public FlowSpec getSpecs(URI uri) throws SpecNotFoundException {
```

Reply:
if you're wondering how Java knows which to choose at runtime, since it's an @Override, per usual it's chosen dynamically based on the runtime type.but it gains more specific typing when that's actually guaranteed statically.
to illustrate, these two are legal:
URI uri = ...
SpecCatalog sc = ...
Spec s = sc.getSpecs(uri);
and
FlowSpecCatalog fsc = ...
FlowSpec fs = fsc.getSpecs(uri);
but this does NOT type-check:
SpecCatalog sc = ...
FlowSpec fs = sc.getSpecs(uri);
make sense?

Reply:
I updated this method but left the other methods that return `Collection<Spec>` intact as to avoid iterating through each one to cast it to a FlowSpec.